internal/pkg/checks/check.go

```diff
@@ -14,41 +14,67 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Ingress conformance test harness machinery
 package checks
 
 import (
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"reflect"
 	"time"
 )
 
+// Config contains test suite configuration fields
 type Config struct {
 }
 
+// Check represents a test case. Checks are named, and must provide a
+// description and a Run function. Checks are organized in a hierarchy.
 type Check struct {
 	Name        string
 	Description string
 
 	Run func(check *Check, config Config) (bool, error)
 
-	// All checks
+	// Child checks
 	checks []*Check
-
 	// Parent check
 	parent *Check
 }
 
+// CapturedRequest contains the original HTTP request metadata as received
+// by the echoserver handling the test request.
+// The DownstreamServiceId field contains the TEST_ID environment variable
+// value of the downstream echoserver.
 type CapturedRequest struct {
-	StatusCode int
-	TestId     string
-	Path       string
-	Host       string
+	DownstreamServiceId string `json:"testId"`
+	Path                string
+	Host                string
+	Method              string
+	Proto               string
+	Headers             map[string][]string
+}
+
+// CapturedResponse contains the HTTP response metadata from the echoserver
+type CapturedResponse struct {
+	StatusCode    int
+	ContentLength int64
+	Proto         string
+	Headers       map[string][]string
 }
 
-func captureRequest(location string, hostOverride string) (data CapturedRequest, err error) {
+// AssertionSet performs checks and accumulates assertion errors
+type AssertionSet []error
+
+func captureRequest(location string, hostOverride string) (capReq CapturedRequest, capRes CapturedResponse, err error) {
```

Comment:
Prefer not naming parameters in functions longer than a few lines :

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters